src/Topics/StandardModernPrecision/Bids1C1Halt.hs

```diff
@@ -0,0 +1,243 @@
+module Topics.StandardModernPrecision.Bids1C1Halt(
+  -- Copied from earlier auctions
+    b1C
+  , b1C1H  -- An alias for b1C1Hnos!
+  , b1C1H1S
+  , b1C1H1N
+  , b1C1H2C
+  , b1C1H2D
+  , b1C1H2H
+  , b1C1H2S
+  -- New bids start here
+  , b1C1H1S1N  -- Responses when opener rebids 1S
+  , b1C1H1S2C
+  , b1C1H1S2D
+  , b1C1H1S2H
+  , b1C1H1S2S
+  , b1C1H1S2N
+
+  , b1C1H2C2D  -- Responses when opener rebids 2C
+  , b1C1H2C2H
+  , b1C1H2C2S
+  , b1C1H2C2N
+  , b1C1H2C3C
+  , b1C1H2C3D
+
+  , b1C1H2D2H  -- Responses when opener rebids 2D
+  , b1C1H2D2S
+  , b1C1H2D2N
+  , b1C1H2D3C
+  , b1C1H2D3D
+  , b1C1H2D3H
+
+  , b1C1H2H2S  -- Responses when opener rebids 2H
+  , b1C1H2H2N
+  , b1C1H2H3C
+  , b1C1H2H3D
+  , b1C1H2H3H
+  , b1C1H2H3S
+) where
+
+import Auction(forbid, suitLength, minSuitLength, maxSuitLength, Action,
+               balancedHand, makeCall, makeAlertableCall, alternatives,
+               longerThan, atLeastAsLong)
+import qualified Terminology as T
+import Topics.StandardModernPrecision.Bids1C(
+    b1C, b1C1Hnos, b1C1H1S, b1C1H1N, b1C1H2C, b1C1H2D, b1C1H2H, b1C1H2S)
+
+-- Alias the bid properly to re-export from here.
+b1C1H :: Action
+b1C1H = b1C1Hnos
+
+
+b1C1H1S1N :: Action
+b1C1H1S1N = do
+    maxSuitLength T.Spades 2
+    mapM_ (`maxSuitLength` 4) T.allSuits
+    makeCall (T.Bid 1 T.Notrump)
+
+
+b1C1H1S2C :: Action
+b1C1H1S2C = do
+    maxSuitLength T.Spades 2
+    minSuitLength T.Clubs 5
+    T.Clubs `longerThan` T.Diamonds
+    T.Clubs `longerThan` T.Hearts
+    makeCall (T.Bid 2 T.Clubs)
+
+
+b1C1H1S2D :: Action
+b1C1H1S2D = do
+    maxSuitLength T.Spades 2
+    minSuitLength T.Diamonds 5
+    T.Diamonds `longerThan` T.Hearts
+    T.Diamonds `atLeastAsLong` T.Clubs
+    makeCall (T.Bid 2 T.Diamonds)
+
+
+b1C1H1S2H :: Action
+b1C1H1S2H = do
+    maxSuitLength T.Spades 2
+    minSuitLength T.Hearts 5
+    T.Hearts `atLeastAsLong` T.Clubs
+    T.Hearts `atLeastAsLong` T.Diamonds
+    makeCall (T.Bid 2 T.Hearts)
+
+
+b1C1H1S2S :: Action
+b1C1H1S2S = do
+    minSuitLength T.Spades 3
+    makeCall (T.Bid 2 T.Spades)
+
+
+b1C1H1S2N :: Action
+b1C1H1S2N = do
+    suitLength T.Clubs 4
+    suitLength T.Diamonds 4
+    suitLength T.Hearts 4
+    suitLength T.Spades 1
+    makeAlertableCall (T.Bid 2 T.Notrump) "1444 shape, singleton spade"
+
+
+b1C1H2C2D :: Action
+b1C1H2C2D = do
+    alternatives . map (`suitLength` 4) $ T.majorSuits
+    balancedHand
+    makeAlertableCall (T.Bid 2 T.Diamonds) "waiting bid"
+
+
+b1C1H2C2H :: Action
+b1C1H2C2H = do
+    minSuitLength T.Hearts 5
+    -- TODO: Do we need extra restrictions here on not having a club fit, or
+    -- having longer hearts than diamonds or anything?
```

Comment:
Your hearts should probably be at least as long as your diamonds. If you had 6 diamonds and 5 hearts, you'd bid the diamonds first, right?